### PR TITLE
Change new metrics config to match Task config and update tests to match

### PR DESCRIPTION
**Pull request by douglas-reid**
I believe the flakiness in `e2e_mixer` presubmit runs to stem from the existing invalid config for the `TestNewMetrics` test (related to bad handler: `handler: handler.prometheus.istio-system`).

This PR attempts to do a couple of things:
1. Use the actual config from the Task for new telemetry again (cannot recall why we moved away)
1. Make the test *fail* if the config isn't used (as opposed to the current situation of *passing* if not used)
1. Add hardcoded delay for config propagation as a placeholder for robustness around config.

Possible additional work:
1. Have the whole send traffic & check for metrics bits be in a retry operation to allow for config slowness.
1. Come up with a better way to determine when Mixer receives the new config

Making PR to enable running the e2e test a bunch of times to determine if/how this improves behavior.

### CODE REVIEW

File: tests/e2e/tests/mixer/mixer_test.go [hunk 1]
```diff
@@ -632,36 +632,57 @@ func TestNewMetrics(t *testing.T) {
...
+		allowPrometheusSync()
+		query := fmt.Sprintf("sum(istio_double_request_count{%s=\"%s\"})", "destination", "productpage-v1")
+		t.Logf("prometheus query: %s", query)
+		value, err := promAPI.Query(ctx, query, time.Now())
+		if err != nil {
+			return fmt.Errorf("could not get metrics from prometheus: %v", err)
+		}
+		t.Logf("promvalue := %s", value.String())
+
+		got, err = vectorValue(value, map[string]string{})
```
Comment (gargnupur) [inline tests/e2e/tests/mixer/mixer_test.go]: can you use queryValue helper function here?

Comment (douglas-reid) [reply tests/e2e/tests/mixer/mixer_test.go]: Done.

File: tests/e2e/tests/mixer/mixer_test.go [hunk 2]
```diff
@@ -621,36 +621,50 @@ func TestNewMetrics(t *testing.T) {
...
+		return nil
 	}
-	log.Infof("promvalue := %s", value.String())
 
-	got, err := vectorValue(value, map[string]string{})
-	if err != nil {
-		t.Logf("prometheus values for istio_response_bytes_count:\n%s", promDump(promAPI, "istio_response_bytes_count"))
-		t.Logf("prometheus values for istio_requests_total:\n%s", promDump(promAPI, "istio_requests_total"))
+	if _, err := retry.Retry(context.Background(), retryFn); err != nil {
+		t.Logf("prometheus values for istio_double_request_count:\n%s", promDump(promAPI, "istio_double_request_count"))
```
Comment (gargnupur) [inline tests/e2e/tests/mixer/mixer_test.go]: This log would be printed twice? once within retryFn and once here...

**Comment by douglas-reid** [comment]
@hklai friendly ping.

**Pull request merged**